R2API.ProcType/ProcTypeAPI.cs

```diff
@@ -0,0 +1,407 @@
+        return result;
+    }
+
+    private static int ProcChainMask_GetHashCode(On.RoR2.ProcChainMask.orig_GetHashCode orig, ref ProcChainMask self)
+    {
+        return orig(ref self) * 397 ^ ModdedMaskHashCode(ProcTypeInterop.GetModdedMask(self));
+    }
+
+    private static bool ProcChainMask_Equals_ProcChainMask(On.RoR2.ProcChainMask.orig_Equals_ProcChainMask orig, ref ProcChainMask self, ProcChainMask other)
+    {
+        return orig(ref self, other) && ModdedMaskEquals(ProcTypeInterop.GetModdedMask(self), ProcTypeInterop.GetModdedMask(other));
+    }
+    #endregion
+
+    #region Internal
+    [SystemInitializer]
+    private static void Init()
+    {
+        buffer = new bool[ModdedProcTypeCount];
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static void NetworkWriteModdedMask(this NetworkWriter writer, bool[] mask)
+    {
+        int maskIndex = 0;
+        for (int i = 0; i < totalByteCount; i++)
+        {
+            byte b = 0;
+            if (mask == null || maskIndex < mask.Length)
+            {
+                int validBitCount = (i < filledByteCount) ? 8 : remainingBitCount;
+                int bitIndex = 0;
+                while (bitIndex < validBitCount)
+                {
+                    if (ArrayUtils.GetSafe(mask, maskIndex))
+                    {
+                        b |= (byte)(1 << bitIndex);
+                    }
+                    bitIndex++;
+                    maskIndex++;
+                }
+            }
+            writer.Write(b);
+        }
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static bool[] NetworkReadModdedMask(NetworkReader reader)
+    {
+        int finalLength = 0;
+        int bufferIndex = 0;
+        for (int i = 0; i < totalByteCount; i++)
+        {
+            int validBitCount = (i < filledByteCount) ? 8 : remainingBitCount;
+            byte b = reader.ReadByte();
+            int j = 0;
+            while (j < validBitCount)
+            {
+                if (buffer[bufferIndex] = (b & (byte)(1 << j)) > 0)
+                {
+                    finalLength = bufferIndex + 1;
+                }
+                j++;
+                bufferIndex++;
+            }
+        }
+        if (finalLength <= 0)
+        {
+            return null;
+        }
+        bool[] result = new bool[finalLength];
+        Array.Copy(buffer, result, finalLength);
+        return result;
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static byte ModdedMaskHashCode(bool[] mask)
+    {
+        byte b = 0;
+        if (mask != null)
+        {
+            int length = Math.Min(mask.Length, 8);
+            for (int i = 0; i < length; i++)
+            {
+                if (mask[i])
+                {
+                    b |= (byte)(1 << i);
+                }
+            }
+        }
+        return b;
+    }
```

Comment:
Kinda weird implementation of hash code. Just straight up ignoring anything after 8th element